### PR TITLE
Change api provider

### PR DESCRIPTION
**Pull request by shnrndk**
Fix: https://github.com/wso2/api-manager/issues/398

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ApisApiServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
+package org.wso2.carbon.apimgt.rest.api.admin.v1.impl;
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ApisApiServiceImpl.java:1]: License header is missing. Please check the other places asa well

Comment (shnrndk) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ApisApiServiceImpl.java:18]: Fixed in https://github.com/wso2/carbon-apimgt/pull/12200/commits/a8775522eeae6a224148469405bb5cdb7c76c4c5

**Review by prasa7 — DISMISSED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/APIConstants.java [hunk 2]
```diff
@@ -244,6 +244,7 @@ public static class Monetization {
     public static final String TAG_COLON_SEARCH_TYPE_PREFIX = "tag:";
     public static final String NAME_TYPE_PREFIX = "name";
     private static final String PROVIDER_SEARCH_TYPE_PREFIX = "provider";
+    public static final String API_PROVIDER_SUFFIX_SLASH = "provider/";
```
Comment (Arshardh) [inline components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/APIConstants.java:247]: Shall we reuse the PROVIDER_SEARCH_TYPE_PREFIX variable here?

Comment (shnrndk) [reply components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/APIConstants.java]: Fixed in https://github.com/wso2/carbon-apimgt/pull/12200/commits/c4e957a25dc551aac6d2e8d6886d4a753b3a2e82

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/SearchApiServiceImpl.java [hunk 3]
```diff
@@ -50,7 +50,10 @@ public class SearchApiServiceImpl implements SearchApiService {
     public Response search(Integer limit, Integer offset, String query, String ifNoneMatch,
                            MessageContext messageContext) throws APIManagementException {
         SearchResultListDTO resultListDTO = new SearchResultListDTO();
-
+        if (!query.equals(RestApiConstants.PIZZASHACK_SEARCH_QUERY) && (query.startsWith("name:") |
```
Comment (Arshardh) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/SearchApiServiceImpl.java:53]: Let's create a git issue to track a better fix for this as we discussed

**Review by Arshardh — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/SearchApiServiceImpl.java (hunk 3 above)
Comment (shnrndk) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/SearchApiServiceImpl.java:53]: Created the git issue

**Pull request merged**